keps/sig-windows/2802-identify-windows-pods-apiserver-admission/README.md

```diff
@@ -0,0 +1,352 @@
+- Interaction with container runtimes.
+
+## Proposal
+
+
+As mentioned [earlier](#Motivation) identifying the Windows pods during the scheduling phase can be done
+in the following ways:
+- Based on [nodeSelector and tolerations](https://kubernetes.io/docs/setup/production-environment/windows/user-guide-windows-containers/#ensuring-os-specific-workloads-land-on-the-appropriate-container-host) in the pod spec with 
+  Windows node specific labels.
+- Based on [runtimeclasses](https://kubernetes.io/docs/setup/production-environment/windows/user-guide-windows-containers/#ensuring-os-specific-workloads-land-on-the-appropriate-container-host) in the pod spec
+
+The problem with using nodeSelector and tolerations is that any unprivileged user/entity can apply the 
+nodeSelector and/or tolerations in the pod spec making it unsecure whereas RuntimeClasses are [recommended](https://kubernetes.io/docs/concepts/containers/runtime-class/#setup) to be
+created by cluster administrator making them authoritative enough to be used.
+
+In order to make `kubernetes.io/os` authoritative enough, kubelet should also reject admitting pods if the pod contains a nodeSelector with label `kubenetes.io/os` with a value not matching `runtime.GOOS`. We can use this check as `kubernetes.io/os` value is supposed to be filled from `runtime.GOOS`. Having this check in kubelet helps in the following conditions:
+- If scheduler has been by passed by setting the nodeName directly on the pod spec.
+- A malicious user or a user unknowingly with RBAC to just bind pod to node, can associate pod to node there by bypassing the scheduler.
+
+
+### User Stories (Optional)
+
+
+#### Story 1
+
+As a Kubernetes cluster administrator, I want appropriate security contexts to be 
+applied to my Windows Pods along with Linux pods
+
+#### Story 2
+
+As a Kubernetes cluster administrator, I want to use my own admission webhook for
+Windows pods.
+
+### Notes/Constraints/Caveats (Optional)
+
+While the `scheduling` field in the RuntimeClass is created to handle scheduling constraints, we have a reserved label
+[`kubernetes.io/os`](https://kubernetes.io/docs/reference/labels-annotations-taints/#kubernetes-io-os) to identify
+Windows nodes from their Linux counterparts. If the RuntimClass's scheduling field has a nodeSelector  `kubernetes.io/os: 'windows'`,
+we can authoritatively say that the pod is indeed a Windows one during the api-server admission time. Querying the RuntimeClass
+during admission time may be expensive operation as the admission plugin is not stateless anymore. 
+
+### Risks and Mitigations
+
+
+
+Existing users may be impacted as we're enforcing to use RuntimeClasses instead of plain nodeSelector and tolerations.
+
+
+The downside of this approach is since we're making RuntimeClass as default choice for the users, existing users have to update their pod spec or workload pod template spec to ensure that their workloads are not broken during upgrades. 
```

Comment:
will we document a prescriptive RuntimeClass?  given RuntimeClass usage varies in the wild, this seems to make all windows workload pod examples hard to follow as they are less portable which is why I keep thinking having a dedicated field on the pod spec itself more portable.